### DISCUSSION TITLE
Can you explain me Marking Mode?

### DISCUSSION THREAD

**Discussion by humanplayer2** [Schneegans/Fly-Pie#174] [Fly-Pie Q&A] [answered]
I may be facing a bug, or I may be slow in uptake. I write here betting on the latter.

I'm trying to go through the tutorial, and for the love of me, I can't get Marking Mode to work for me. 

I've tried to LMB click the center, hold LMB and move to the fire, pause, and go straight down, and I've tried LMB clicking the fire directly, hold LMB, and move down, but I nothing happens until I release LMB again..

Am I doing something wrong?

**Comment by Schneegans** [comment, marked as answer]
Hi there!

I just tested it on GNOME Shell 3.38 and everything seems to work as supposed on my end. I never tested it on Pop!_OS though. What you attempted seems to be alright... Basically, you have to press the mouse button anywhere (it doesn't really matter if you hit the center or an item) and then drag the mouse outwards. As long as you keep moving your mouse inside an item's wedge, the corresponding item will be "attached" to your pointer. As soon as you stop the movement for a fraction of a second or make a sharp turn, the item will be selected (you do not have to release the button). However, final leaf items which are no submenus will only be selected if you release the button. Here's a short clip:  

https://user-images.githubusercontent.com/829942/144708873-2ad94cf1-75dc-4aac-9af7-fcf783bba691.mp4

The green highlight on the mouse cursor appears whenever the left mouse button is pressed. The only pitfall I can think of is that there's a minimum distance you have to move the pointer away from the center for items to become selected. You can see this in the beginning of the clip where I drag items around close to the center without selecting them.

**Reply by ggerganov** [reply]
I am experiencing the same problem as explained by @humanplayer2 :

https://user-images.githubusercontent.com/1991296/148802008-0a1b7b97-0b42-44cb-ba64-88f06e1b0316.mp4

System info:

```
> gnome-shell --version
GNOME Shell 3.36.9

> lsb_release --all
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```

**Reply by Schneegans** [reply]
I just tested it and apparently that's a bug! It just occurs on GNOME 3.36 and 3.38, that's why I haven't noticed it before. I opened #200 accordingly. It will not easy to be fixed, though...

**Comment by Schneegans** [comment]
Another thing, which might have happened, is that you tweaked the advanced settings by accident. For example, this minimum distance is configurable and maybe you have set it to a large value? Here are the defaults:
![Screenshot from 2021-12-04 13-22-42](https://user-images.githubusercontent.com/829942/144709263-e9b7e91a-53d6-4007-afc3-ff06fa630ea9.png)

**Comment by humanplayer2** [comment]
Thank you very much for the video. It verifies that I had understood things correctly from the tutorial. Alas, it doesn't work. I also seem to face an issue with "Bind to Super + RMB" which doesn't open the menu. I'll open an issue.

(Turbo-Mode works).

**Comment by Schneegans** [comment]
I just pushed something to the branch `feature/tryfix_200` which may fix this issue. I did some quick tests and did not observe any bugs, but more in-depth testing is necessary as this may affect all interaction modes. The bad thing is, that this may also have an impact on the touch support but I do not have a touch device at hand to test this...

There is [a guide available](https://github.com/Schneegans/Fly-Pie/blob/develop/docs/installation.md#c-cloning-the-latest-version-with-git) for installing the latest version from git. Just do a `git checkout feature/tryfix_200` before `make install`.

**Reply by ggerganov** [reply]
It works now:

![image](https://user-images.githubusercontent.com/1991296/149089251-012a4d67-e8b0-4a2a-ba81-8d5dfedd3dfb.png)

**Reply by Schneegans** [reply]
Great :laughing:. Hopefully I find someone with a touch device who could test this as well. Then we should release this ASAP!

**Reply by humanplayer2** [reply]
Thank you very much for this! I will re-activate and earn some achievements!

**Reply by Schneegans** [reply]
@VinnyBoiler confirmed that touch seems to be working just fine with #201. Therefore, I will merge this and release a small update later today.
